packages/agreement-process/test/createAgreement.test.ts

```diff
@@ -201,13 +207,85 @@ describe("create agreement", () => {
     await addOneTenant(consumer);
     await addOneEService(eservice);
 
-    const apiAgreementPayload: agreementApi.AgreementPayload = {
-      eserviceId: eservice.id,
-      descriptorId: eservice.descriptors[0].id,
+    const createdAgreement = await agreementService.createAgreement(
+      {
+        eserviceId: eservice.id,
+        descriptorId: eservice.descriptors[0].id,
+      },
+      {
+        authData,
+        correlationId: generateId(),
+        serviceName: "",
+        logger: genericLogger,
+      }
+    );
+
+    await expectedAgreementCreation(
+      createdAgreement,
+      eservice.id,
+      descriptor.id,
+      eserviceProducer.id,
+      consumer.id
+    );
+  });
+  it("should succeed when the delegationId is provided, the requester is the delegated and the delegator has all Descriptor certified Attributes not revoked", async () => {
+    const authData = getRandomAuthData();
+    const eserviceProducer: Tenant = getMockTenant();
+
+    const certifiedDescriptorAttribute1: EServiceAttribute =
+      getMockEServiceAttribute();
+    const certifiedDescriptorAttribute2: EServiceAttribute =
+      getMockEServiceAttribute();
+
+    const descriptor = getMockDescriptorPublished(generateId<DescriptorId>(), [
+      [certifiedDescriptorAttribute1],
+      [certifiedDescriptorAttribute2],
+    ]);
+
+    const certifiedTenantAttribute1: TenantAttribute = {
+      ...getMockCertifiedTenantAttribute(certifiedDescriptorAttribute1.id),
+      revocationTimestamp: undefined,
     };
 
+    const certifiedTenantAttribute2: TenantAttribute = {
+      ...getMockCertifiedTenantAttribute(certifiedDescriptorAttribute2.id),
+      revocationTimestamp: undefined,
+    };
+
+    const delegator = getMockTenant(authData.organizationId, [
```

Comment:
Generate a new id for the delegator, otherwise they are the same tenant

```suggestion
    const delegator = getMockTenant(generateId<TenantId>(), [
```